update ocaml-native to 4.02.3

starius opened the pull request:
See https://lists.nongnu.org/archive/html/mingw-cross-env-list/2015-08/msg00003.html

starius commented on src/ocaml-native-1-fixes.patch:
```diff
@@ -24,78 +27,70 @@ index b5ef878..77454ed 100644
+    "-install-bin-dir", Set_string Ocamlbuild_where.bindir, "<path> Set the install binary directory";
+    "-where", Unit (fun () -> print_endline !Ocamlbuild_where.libdir; raise Exit_OK), " Display the install library directory";
+-   "-which", String (fun cmd -> print_endline (find_tool cmd); raise Exit_OK), "<command> Display path to the tool command";
++   "-which", String (fun cmd -> print_endline (find_tool cmd "FIXME"); raise Exit_OK), "<command> Display path to the tool command";
```
Please review this line carefully. Is it OK to replace extra argument with string "FIXME"?

tonytheodore replied:
I think so, I don't know anything about ocaml but an updated version with minor issues is better than an outdated version.

starius commented:
Package ocaml-core must be updated as well. It shares ocaml version with ocaml-native.

tonytheodore commented:
From #1003, this should also be moved to the native target so it's built only once. It's [building on OSX now](https://github.com/mxe/mxe/commit/a96fcf41143baa5aaf3c0e31a9fe204089fad3d9) so will be easier for me to test.

tonytheodore commented:
Also, project has moved:

http://ocaml.org/
https://github.com/ocaml/ocaml

I wonder if we should make this a plugin?

P.S. @starius, could you update the checksum on this and rebase?

starius commented:
Rebased, updated checksum, patch consolidated.

It doesn't work. I failed to update crosscompiled ocaml. Somebody who knows ocaml should look at it.

> I wonder if we should make this a plugin?

Why?

tonytheodore commented:
See #2512
